enumerables.rb

```diff
@@ -0,0 +1,176 @@
+# rubocop: disable Metrics/ModuleLength
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    return to_enum :my_each unless block_given?
+
+    i = 0
+    while i < size
+
+      case self.class.name
+      when 'Hash' then yield(keys[i], self[keys[i]])
+      when 'Array' then yield(self[i])
+      when 'Range' then yield(to_a[i])
+      end
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
```

Comment:
* [ ] There should be a working `my_each_with_index` identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

### replace ```my_each_with_index``` with ```each_with_index``` and compare results in the example below.
``` p [1,2,3,4,5].my_each_with_index{ |a,b| b} ```